src/middleware.ts

```diff
@@ -1,14 +1,16 @@
 import { authMiddleware } from "@clerk/nextjs";
 import { redirectToSignIn } from "@clerk/nextjs/server";
 import { NextRequest, NextResponse } from "next/server";
+import { clerkClient } from "@clerk/nextjs/server";
 
 // This example protects all routes including api/trpc routes
 // Please edit this to allow other routes to be public as needed.
 // See https://clerk.com/docs/references/nextjs/auth-middleware for more information about configuring your Middleware
 export default authMiddleware({
   publicRoutes: ["/"],
   afterAuth: async (auth, req: NextRequest) => {
-    const { userId, sessionClaims } = auth
+    const { userId } = auth
+    const user = userId && await clerkClient.users.getUser(userId);
```

Comment:
Using `clerkClient` like this is Middleware is definitely not something that should be done for two reasons:
1. This introduces a network request and the related latency to all route view/page loads.
2. This introduces the frequent use of a BAPI endpoint, which is rate limited. This kind of use would mean that a moderately busy app with 100 route views in 10 seconds would then start getting rate limiting. That also doesn't account for any other users of `currentUser` or `clerkClient` in the application.